substrate/runtime/primitives/src/bft.rs

```diff
@@ -111,17 +110,20 @@ impl Slicable for Action {
 	}
 }
 
+/// Type alias for extracting message type from block.
+pub type MessageFor<B> = Message<B, <B as ::traits::Block>::Hash>;
```

Comment:
What does "message" mean here? What if the block type doesn't have the notion of "message"?

Reply:
Since it's in the `bft` module, it's fair to assume that this is a BFT message. The BFT subsystem is something that operates over block types as opposed to being a property of them.